packages/web-components/src/popover/popover.ts

```diff
@@ -0,0 +1,292 @@
+
+    // ref might have changed -> force restart
+    this.handlePositioningStartStop(true);
+
+    console.groupEnd();
+  }
+
+  @observable
+  public anchorRef?: HTMLElement[];
+
+  protected anchorRefChanged() {
+    console.group(this.objId, 'anchorRefChanged');
+
+    // ref might have changed -> force restart
+    this.handlePositioningStartStop(true);
+
+    console.groupEnd();
+  }
+
+  /**
+   * @internal
+   * List of child popovers nested inside this popover.
+   * Used to avoid closing this popover on ESC key press when a child popover is open.
+   */
+  private childPopovers: Popover[] = [];
+
+  /**
+   * @internal
+   * Called by child popovers in their connectedCallback to register themselves.
+   */
+  protected registerChildPopover(popover: Popover) {
+    this.childPopovers.push(popover);
+  }
+
+  /**
+   * @internal
+   * Called by child popovers in their disconnectedCallback to unregister themselves.
+   */
+  protected unregisterChildPopover(popover: Popover) {
+    this.childPopovers = this.childPopovers.filter(x => x !== popover);
+  }
+
+  /**
+   * @internal
+   * Traverses the DOM tree upwards, crossing shadow root boundaries, to find the first parent Popover.
+   */
+  private getParentPopover(): Popover | undefined {
+    let parent = this.parentNode;
+    while (parent) {
+      console.log(parent);
+      if (parent instanceof Popover) {
+        return parent;
+      }
+      parent = parent.parentNode ?? (parent as ShadowRoot).host;
+    }
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+
+    const parentPopover = this.getParentPopover();
+    if (parentPopover) {
+      parentPopover.registerChildPopover(this);
+    }
+    console.group(this.objId, 'connectedCallback', { parentPopover });
+
+    // the component might have been mounted with `open` set to true
+    this.handlePositioningStartStop();
+
+    console.groupEnd();
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    console.group(this.objId, 'disconnectedCallback');
+
+    this.handlePositioningStartStop();
+
+    const parentPopover = this.getParentPopover();
+    if (parentPopover) {
+      parentPopover.unregisterChildPopover(this);
+    }
+
+    console.groupEnd();
+  }
+
+  private isPositioningStarted = false;
+  private autoUpdateCleanup: (() => void) | undefined;
+  private handlePositioningStartStop(forceRestart: boolean = false) {
+    const shouldStart =
+      (this as FASTElement).$fastController.isConnected && this.open && this.popoverContentRef && !!this.anchorRef?.[0];
+
+    console.log(this.objId, 'handlePositioningStartStop', {
+      open: this.open,
+      connected: (this as FASTElement).$fastController.isConnected,
+      anchorRef: this.anchorRef,
+      popoverContentRef: this.popoverContentRef,
+      shouldStart,
+      positioningStarted: this.isPositioningStarted,
+      forceRestart,
+    });
+
+    if (shouldStart) {
```

Comment:
Do you need this check it seems that either way ` if (this.isPositioningStarted)` then you call `this.stopPositioning`